ReactNative/ReactTwitter/core/service/oauth-helper.js

```diff
@@ -0,0 +1,75 @@
+import HmacSHA1 from 'crypto-js/hmac-sha1'
+import CryptoJS from 'crypto-js'
+import PercentEncoder from './percent-encoder.js'
+
+class OauthHelper {
+  constructor (consumerSecret) {
+    this.consumerSecret = consumerSecret
+  }
+
+  buildAuthorizationHeader (method, url, params, oauthTokenSecret) {
+    let output = 'OAuth '
+    let signature = this.getSigningKey(method, url, params, oauthTokenSecret)
+    for (let key in params) {
+      output += PercentEncoder.encode(key) + '="' + PercentEncoder.encode(params[key]) + '", '
+    }
+    output += 'oauth_signature="' + PercentEncoder.encode(signature) + '"'
+    return output
+  }
+
+  getSigningKey (method, url, params, oauthTokenSecret) {
+    let signingKey = PercentEncoder.encode(this.consumerSecret) + '&' + PercentEncoder.encode(oauthTokenSecret)
+    let signatureBase = OauthHelper._getSignatureBase(method, url, params)
+    return CryptoJS.enc.Base64.stringify(HmacSHA1(signatureBase, signingKey))
+  }
+
+  static _getSignatureBase (method, url, params) {
```

Comment:
To be honest, I wanted to avoid instantiating the class for the test, so yes 👅 
But either way, it's not using any object fields, so I don't see a reason why this shouldn't be static. Is it a bad practice?